### PR TITLE
Integration performances

### PR DESCRIPTION
**Pull request by AbbasSarwar**
### Features
* Performances improved N + 1
* Tests created for Interface
* Readme Updated

rspec spec/features
Performances Before:
before (1)
before (2)

Performances After:
After (1)

### CODE REVIEW

File: spec/features/users_system_spec.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                            post_counter: 0)
+  end
+  describe 'user index' do
+    before(:each) do
+      visit users_path
+    end
+    it 'shows the user details' do
+      expect(page).to have_content(@profile1.name)
+      expect(page).to have_content(@profile2.name)
+    end
```
Comment (Davidosky007) [inline spec/features/users_system_spec.rb]: * Kindly add these missing tests shown below 👇  as they are [required](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=User%20post%20index,each%20commentor%20left.) 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @AbbasSarwar @emhamza 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* N+1 issue was fixed 👍 
* Well written PR 👍 
* No linter errors 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/users_system_spec.rb [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    it 'who commented?' do
+      expect(page).to have_content(@profile1.name)
+    end
+    it 'show all comments with name:' do
+      expect(page).to have_content 'Tom: Hello'
+      expect(page).to have_content 'Tom: wassup'
+    end
+  end
+end
```
Comment (TedLivist) [inline spec/features/users_system_spec.rb:158]: - [ ] **[OPTIONAL]** These posts tests should be in their own files to help create context and organization

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @AbbasSarwar and @emhamza,

✔️ Well-written and passing tests

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/users_system_spec.rb (hunk 2 above)
Comment (AbbasSarwar) [reply spec/features/users_system_spec.rb:158]: thank you I will try to rearrange these files.

**Pull request merged**